app/validators/upload_validator.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class UploadValidator < ActiveModel::Validator
+  def validate(upload)
+    @upload = upload
+
+    validate_search_engine
+    validate_csv_file
+  end
+
+  private
+
+  attr_reader :upload
+
+  def search_engine
+    upload.search_engine
+  end
+
+  def csv_file
+    upload.csv_file
+  end
+
+  def validate_search_engine
```

Comment:
All these validations could be defined in https://github.com/nimblehq/ic-rails-huey-toby/pull/31/files#diff-becd1596dd6a914113c0a2b7b5cde93d201b7bfeb4c5a9ca27b96950ce329339R5 using `ActiveModel::Validations` DSL. Usually, we write a custom, validator for 1) a single validation (not multiple) and 2) for validations that we cannot do with the built-in DSL.